packages/service-library/src/servicelib/archiving_utils/_interface_7zip.py

```diff
@@ -0,0 +1,330 @@
+import asyncio
+import asyncio.subprocess
+import logging
+import os
+import re
+from collections.abc import Awaitable, Callable
+from contextlib import AsyncExitStack
+from pathlib import Path
+from typing import Final
+
+import tqdm
+from models_library.basic_types import IDStr
+from pydantic import NonNegativeInt
+from servicelib.logging_utils import log_catch
+from tqdm.contrib.logging import tqdm_logging_redirect
+
+from ..file_utils import shutil_move
+from ..progress_bar import ProgressBarData
+from ._errors import ArchiveError
+from ._tdqm_utils import (
+    TQDM_FILE_OPTIONS,
+    TQDM_MULTI_FILES_OPTIONS,
+    compute_tqdm_miniters,
+    human_readable_size,
+)
+from ._utils import iter_files_to_compress
+
+_logger = logging.getLogger(__name__)
+
+_TOTAL_BYTES_RE: Final[str] = r" (\d+)\s*bytes"
+_FILE_COUNT_RE: Final[str] = r" (\d+)\s*files"
+_PROGRESS_PERCENT_RE: Final[str] = r" (?:100|\d?\d)% "
+_ALL_DONE_RE: Final[str] = r"Everything is Ok"
+
+_7ZIP_PATH: Final[Path] = Path("/usr/bin/7z")
+
+_TABLE_HEADER_START: Final[str] = "------------------- "
+_FILE_PERMISSIONS: Final[str] = " ..... "
```

Comment:
what are these? comparison strings? it does not look like a file permissions to me